--- cmake-2.8.12/Modules/FindImageMagick.cmake.0006~	2013-12-27 01:31:37.077298974 +0100
+++ cmake-2.8.12/Modules/FindImageMagick.cmake	2013-12-27 01:40:28.700639284 +0100
@@ -73,12 +73,16 @@
 # Helper functions
 #---------------------------------------------------------------------
 function(FIND_IMAGEMAGICK_API component header)
+  find_package(PkgConfig QUIET)
+  pkg_check_modules(PC_ImageMagick_${component} ${component})
   set(ImageMagick_${component}_FOUND FALSE PARENT_SCOPE)
 
   find_path(ImageMagick_${component}_INCLUDE_DIR
     NAMES ${header}
     PATHS
       ${ImageMagick_INCLUDE_DIRS}
+      ${PC_ImageMagick_${component}_INCLUDEDIR}
+      ${PC_ImageMagick_${component}_INCLUDE_DIRS}
       "[HKEY_LOCAL_MACHINE\\SOFTWARE\\ImageMagick\\Current;BinPath]/include"
     PATH_SUFFIXES
       ImageMagick ImageMagick-6
