diff -urN cmake-2.8.9/Modules/FindFLTK.cmake cmake-2.8.9-patched/Modules/FindFLTK.cmake
--- cmake-2.8.9/Modules/FindFLTK.cmake	2012-08-10 05:15:19.000000000 +1100
+++ cmake-2.8.9-patched/Modules/FindFLTK.cmake	2012-09-05 22:15:19.522508032 +1100
@@ -99,6 +99,7 @@
     ${FLTK_DIR_SEARCH}
 
     # Look in standard UNIX install locations.
+    /usr/lib/FLTK-1.1
     /usr/local/lib/fltk
     /usr/lib/fltk
     /usr/local/fltk
